beacon/beacon_test.go

```diff
@@ -119,9 +119,8 @@ func createATX(
 		types.NIPostChallenge{PublishEpoch: lid.GetEpoch()},
 		types.GenerateAddress(types.RandomBytes(types.AddressLength)),
 		numUnits,
-		&nonce,
 	)
-
+	atx.VRFNonce = nonce
```

Comment:
If we want to make sure that `atx.VRFNonce` is never incorrectly `0` `types.NewActivationTx` should take it as a parameter.

Otherwise if some code creates a `types.ActivationTx` and then doesn't set `VRFNonce` to some value it is an incorrect `0`. Wdyt?

Reply:
The `types.NewActivationTx()` is only used in tests. Most of the tests passed `nil`, as they don't care about the VRF nonce anyway. I would need to update them to pass a value (zero, which is also correct) in the place where they passed nil so far.

Following this logic, `types.NewActivationTx()` would need to take values for and set all fields, which I think defeats the point of such a helper.

Reply:
We could get rid of this function completely to avoid this issue. If we do not want to blow up this PR, maybe we just add a comment to `types.NewActivationTx` that it shouldn't be used outside of tests for now? 🙂